server/mailrepository/mailrepository-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryPath.java

```diff
@@ -29,21 +29,42 @@
 import com.google.common.base.Preconditions;
 
 public class MailRepositoryPath implements Comparable<MailRepositoryPath> {
-    public static final MailRepositoryPath fromEncoded(String encodedPath) throws UnsupportedEncodingException {
-        return new MailRepositoryPath(URLDecoder.decode(encodedPath, StandardCharsets.UTF_8.displayName()));
+
+    private static final String PATH_DELIMITER = "/";
+
+    public static MailRepositoryPath fromEncoded(String encodedPath) throws UnsupportedEncodingException {
+        Preconditions.checkNotNull(encodedPath, "Supplied MailRepositoryPath valuie is null");
+
+        return from(URLDecoder.decode(encodedPath, StandardCharsets.UTF_8.displayName()));
     }
 
-    public static final MailRepositoryPath from(String path) {
-        return new MailRepositoryPath(path);
+    public static MailRepositoryPath from(String path) {
+        Preconditions.checkNotNull(path, "Supplied MailRepositoryPath valuie is null");
+
+        return new MailRepositoryPath(sanitizePath(path));
+    }
+
+    private static String sanitizePath(String path) {
+        if (!path.endsWith(PATH_DELIMITER)) {
+            return path;
+        }
+        return sanitizePath(path.substring(0, path.length() - 1));
```

Comment:
is it a "strip last char" method?

Reply:
Sounds like it yes

Reply:
```
    private static String sanitizePath(String path) {
        return StringUtils.stripEnd(path, "/");
    }
```

Makes it even more simple and avoids the recursion...